File: exercises/ba1h.py
from library.algorithms import pattern_find_approx


def ba1h(params: list[str]):
    pattern, text, d = params
    d = int(d)
    return pattern_find_approx(text, pattern, d)


def test_ba1h(benchmark):
    expected = [6, 7, 26, 27, 78]
    result = benchmark(ba1h, ["ATTCTGGA", "CGCCCGAATCCAGAACGCATTCCCATATTTCGGGACCACTGGCCTCCACGGTACGGACGTCAATCAAATGCCTAGCGGCTTGTGGTTTCTCCTACGCTCC", "3"])
    assert expected == result

File: exercises/ba1h.go
package exercises

import (
	"log"
	"strconv"

	"github.com/vgalaktionov/bioinformatics/lib"
)

func (ex Exercise) BA1H(params []string) []int {
	pattern := params[0]
	text := params[1]
	d, err := strconv.Atoi(params[2])
	if err != nil {
		log.Fatalf("%s", err)
	}

	result := lib.PatternFindApprox(text, pattern, d)

	return result
}
